lib/v2/screens/claim_invite_code/claim_invite_code_screen.dart

```diff
@@ -0,0 +1,126 @@
+import 'package:seeds/providers/notifiers/settings_notifier.dart';
+import 'package:seeds/v2/components/flat_button_long.dart';
+import 'package:seeds/v2/components/scanner/seeds_qr_code_scanner_widget.dart';
+import 'package:seeds/v2/components/text_form_field_light_custom.dart';
+import 'package:seeds/design/app_theme.dart';
+import 'interactor/claim_invite_code_bloc.dart';
+
+class ClaimInviteCodeScreen extends StatefulWidget {
+  const ClaimInviteCodeScreen({
+    Key key,
+  }) : super(key: key);
+
+  @override
+  _ClaimInviteCodeScreenState createState() => _ClaimInviteCodeScreenState();
+}
+
+class _ClaimInviteCodeScreenState extends State<ClaimInviteCodeScreen> {
+  ClaimInviteCodeBloc _ClaimInviteCodeBloc;
+  final _keyController = TextEditingController();
+  final GlobalKey _qrKey = GlobalKey(debugLabel: 'QR');
+
+  @override
+  void initState() {
+    super.initState();
+    _ClaimInviteCodeBloc = ClaimInviteCodeBloc(SettingsNotifier.of(context), AuthNotifier.of(context));
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider(
+      create: (BuildContext context) => _ClaimInviteCodeBloc,
+      child: Scaffold(
+        appBar: AppBar(),
+        body: SingleChildScrollView(
+          child: Column(
+            mainAxisAlignment: MainAxisAlignment.end,
+            children: [
+              const SizedBox(
+                height: 60,
+              ),
+              const Text(
+                "Scan QR Code",
+                textAlign: TextAlign.center,
+              ),
+              const SizedBox(
+                height: 50,
+              ),
+              SeedsQRCodeScannerWidget(
+                onQRViewCreated: _onQRViewCreated,
+                qrKey: _qrKey,
+              ),
+              const SizedBox(
+                height: 75,
+              ),
+              Container(
+                decoration: const BoxDecoration(
+                    borderRadius: BorderRadius.only(topRight: Radius.circular(30), topLeft: Radius.circular(30)),
+                    color: AppColors.white),
+                child: Padding(
+                  padding: const EdgeInsets.only(left: 20, right: 20),
+                  child: Form(
+                    child: Column(
+                      crossAxisAlignment: CrossAxisAlignment.start,
+                      children: [
+                        const SizedBox(
+                          height: 20,
+                        ),
+                        Text(
+                          'Or enter by yourself below',
+                          style: Theme.of(context).textTheme.subtitle2Black,
+                          textAlign: TextAlign.end,
+                        ),
+                        const SizedBox(
+                          height: 15,
+                        ),
+                        TextFormFieldLightCustom(
+                          controller: _keyController,
+                          validator: (value) {
```

Comment:
No, all validation goes inside the bloc. 
We dont need a _keyController or a validator, all the logic goes inside the bloc

Reply:
By passing the content to the validator you can obtain a native field validation.
If there is an error it will print a red text below the textfield. I think this is the easiest unless you want to manually manage the error by creating a conditioned text in the screeen managed from the Bloc.

Also this would be the only case to use the Bloc to manage validations if you have more than one textField you by force you need to use the Form and a global key to validate all the fields.

Reply:
I agree with Rauls comment

Reply:
No, All logic goes inside the blocs. No logic in the UI

Reply:
This case of scenaries I personally handle by create a StringField class in order to encapsulate and extend validations for any String.

So from bloc State you can define:

```
      privateKey: StringField(
        isRequired: true,
        requiredErrorMessage: "Private Key cannot be empty",
        label: 'Private key*',
        placeHolder: 'Apellidos',
        minLengthRequired: 18,
        minLengthErrorMessage: 'Privete key must be at less 10 caracteres',
        maxLengthRequired: 50,
      ),
```
And in the UI you can call a instance function or getter whatever:
```
                         validator: (_) {
                            return state.privateKey.isValid
                                ? null
                                : state.privateKey.error;
                          },
```
In the Bloc you need return a new instance on value change 
```
    if (event is privateKeyChanged) {
      var newPrivateKey = state.privetKey.copyWith(value: event.privateKey);
      yield state.copyWith(privateKey: newPrivateKey);
    }
```

Reply:
If you want to go further (with the purpose of having the ui cleaner) you can even make isValid return null or message.

`validator: (_) = > state.privateKey.isValid,`

Reply:
So what do you thing of this approach? @gguijarro-c-chwy  I can made a PR with this utility class, I would put it in domain shared so @gguij004 I can use it.